airqo-platform/AirQo-api
# Kafka calibrate

# Feature

**_WHAT DOES THIS PR DO?_**
Creates and integrates a deployment of the calibrate model into the measurements pipeline.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
https://github.com/airqo-platform/AirQo-frontend/issues/321

**_WHAT IS THE LINK TO THE PR BRANCH_**
[kafka-calibrate](https://github.com/airqo-platform/AirQo-api/tree/kafka-calibrate)

**_HOW DO I TEST OUT THIS PR?_**
Add file [airqo-250220-5149c2aac8f2.json](https://drive.google.com/file/d/1X5-BMtgv2UzSV7Y0_mTrHmIsfyFQT08q/view?usp=sharing) to `jobs/`.
[.env file](https://docs.google.com/document/d/1nu95orw9tJKu3wfriPD7z_l2SsQElTLA0qw4ZR3UMxs/edit?usp=sharing)
Create a virtual environment
```
cd AirQo-api/src/calibrate
python3 -m venv [DESTINATION_FOLDER]
source [DESTINATION_FOLDER]/bin/activate
pip install -r requirements.txt
python kafka_consumer.py
```
You should see outputs of measurements with calibrated values

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
None

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
None

**_ARE THERE ANY RELATED PRs?_**
None
